src/main/java/org/minions/devfund/marylin/WhackAMole.java

```diff
@@ -0,0 +1,174 @@
+
+        for (char[] row : this.moleGrid) {
+            Arrays.fill(row, '*');
+        }
+    }
+
+    /**
+     * Method to place a mole into the grid.
+     *
+     * @param x X axis in the grid.
+     * @param y Y axis in the grid.
+     * @return Returns true if the mole was place, false otherwise.
+     */
+    private boolean place(int x, int y) {
+        if (this.moleGrid[x][y] != 'M') {
+            this.moleGrid[x][y] = 'M';
+            this.molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Method to whack a mole in the grid.
+     *
+     * @param x X axis in the grid.
+     * @param y Y axis in the grid.
+     */
+    private void whack(int x, int y) {
+        if (this.moleGrid[x][y] == 'M') {
+            this.moleGrid[x][y] = 'W';
+            this.score++;
+            this.molesLeft--;
+        }
+        this.attemptsLeft--;
+    }
+
+    /**
+     * Method to print the status of the game without shows the moles.
+     */
+    private void printGridToUser() {
+        for (char[] rowGrid : this.moleGrid) {
+            StringBuilder strBuilderRow = new StringBuilder();
+            for (char charItem : rowGrid) {
+                if (charItem == 'M') {
+                    strBuilderRow.append("*  ");
+                } else {
+                    strBuilderRow.append(charItem);
+                    strBuilderRow.append("  ");
+                }
+            }
+            LOGGER.info(strBuilderRow.toString());
+        }
+    }
+
+
+    /**
+     * Method to print the grid showing the status of the game even moles.
+     */
+    private void printGrid() {
+        for (char[] rowGrid : this.moleGrid) {
+            StringBuilder strBuilderRow = new StringBuilder();
+            for (char charItem : rowGrid) {
+                strBuilderRow.append(charItem);
+                strBuilderRow.append("  ");
+            }
+            LOGGER.info(strBuilderRow.toString());
+        }
+    }
+
+    /**
+     * Main method to play the game.
+     *
+     * @param args Default Argument.
+     */
+    public static void main(final String[] args) {
+        int numberOfMoles = 10;
+        int numAttemps = 50;
+        int gridLenght = 10;
+
+        //create a 10 by 10 grid
+        WhackAMole whackAMole = new WhackAMole(numAttemps, gridLenght);
+
+        //Place randomly a total of 10 moles
+        for (int i = 0; i < numberOfMoles; i++) {
+            Random rand = new Random();
+            int x = rand.nextInt(numberOfMoles);
+            int y = rand.nextInt(numberOfMoles);
+            StringBuilder strBuilderMessage = new StringBuilder();
+            if (!whackAMole.place(x, y)) {
+                strBuilderMessage.append("There is already a mole at ");
+                strBuilderMessage.append(x);
+                strBuilderMessage.append(",");
+                strBuilderMessage.append(y);
+                strBuilderMessage.append(". Trying again!");
+                i--;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor the code in order to not assign to this loop counter from within the loop body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AForLoopCounterChangedCheck)